lib/spack/spack/concretize.py

```diff
@@ -70,15 +69,8 @@ class Concretizer:
     concretization strategies, or you can override all of them.
     """
 
-    #: Controls whether we check that compiler versions actually exist
-    #: during concretization. Used for testing and for mirror creation
-    check_for_compiler_existence = None
-
     def __init__(self, abstract_spec=None):
-        if Concretizer.check_for_compiler_existence is None:
```

Comment:
This was rather questionable... __init__ as a static initializer for a global :|

Reply:
lmao, it was even worse, `from spack.config import config` fixing the binding on import time